lib/phobos/deep_struct.rb

```diff
@@ -0,0 +1,37 @@
+module Phobos
+  class DeepStruct < OpenStruct
+    # Based on
+    # https://docs.omniref.com/ruby/2.3.0/files/lib/ostruct.rb#line=88
+    def initialize(hash=nil)
+      @table = {}
+      @hash_table = {}
+
+      if hash
+        hash.each_pair do |k, v|
+          k = k.to_sym
```

Comment:
This DeepStruct class is only intended for config, but the name suggests it's more general, so it may find itself used on incoming data in the future. Historically, `k.to_sym` could have opened up DoS holes. Do we care about this post Ruby 2.2?